subprojects/docs/src/docs/userguide/console.adoc

```diff
@@ -0,0 +1,147 @@
+
+The following portion of the console output shows the work in-progress display in action:
+
+----
+<==========---> 77% EXECUTING [10s]
+> :codeQuality:classpathManifest > Resolve dependencies :codeQuality:runtimeClasspath
+> :ivy:classpathManifest > Resolve dependencies :ivy:runtimeClasspath
+> IDLE
+> :antlr:classpathManifest > Resolve dependencies :antlr:runtimeClasspath
+> :scala:compileJava > Resolve dependencies :scala:compileClasspath
+> :buildInit:classpathManifest > Resolve dependencies :buildInit:runtimeClasspath
+> :jacoco:classpathManifest > Resolve dependencies :jacoco:runtimeClasspath
+> IDLE
+----
+
+[[sec:console_build_result]]
+==== Build result
+
+After finishing the build, the console will render the outcome of the build and a ratio of work performed vs. work avoided. The outcome indicates if the build was successful or failed. It also displays the elapsed time it took to run the overall build. The work ratio gives the user an idea about the number of performed tasks or tasks that did not need to be executed. The intention of the summarized view is to avoid too much information in the console output during the build.
+
+----
+BUILD SUCCESSFUL in 2m 10s
+411 actionable tasks: 381 executed, 30 up-to-date
+----
+
+[[sec:console_build_output]]
+==== Build output
+
+Build output is rendered right above the build progress bar. Typical output includes log messages, messages from the standard output and error streams, compile warnings, test output events and many more.
+
+Starting with version 4.0 of Gradle, the console output has been reduced dramatically. The console does not render every task with its build status anymore - it only renders output that is emitted during a build lifecycle phase or a task execution. Avoiding an abundance of information keeps output as readable as possible. Gradle grouped output originating from a specific context e.g. all warnings from a compilation task. Grouped output is especially useful in a project setup that enables parallel project execution as it prevents concurrent interleaved messages to be interleaved without clearly indicating its origin (see xref:parallel_execution[]).
+
+NOTE: To render console output with the formatting of pre-4.0 versions of Gradle, add the <<gradle_command_line,command line>> option `--console=plain`.
+
+The following console output shows grouped output for the configuration phase and the task `:compileJava`:
+
+----
+> Configure project :
+Configuring project version.
+Using a single directory for all classes from a source set. This behaviour has been deprecated and is scheduled to be removed in Gradle 5.0
```

Comment:
The deprecation warning here probably doesn't make sense.  I would do something more plainly obvious like:

```
> Configure project ':library'
Configuring project version for project ':library'
> Configure project ':consumer'
Configuring project version for project ':consumer'
```